javascript/javascript2/week2/movies.js

```diff
@@ -0,0 +1,162 @@
+    running_times: 5820,
+  },
+  {
+    title: "10 Years",
+    year: 2011,
+    rating: 6.1,
+    votes: 20790,
+    running_times: 6000,
+  },
+  {
+    title: "10,000 BC",
+    year: 2008,
+    rating: 5.1,
+    votes: 114750,
+    running_times: 6540,
+  },
+  {
+    title: "100 Feet",
+    year: 2008,
+    rating: 5.5,
+    votes: 10979,
+    running_times: 6240,
+  },
+  {
+    title: "100 Girls",
+    year: 2000,
+    rating: 5.9,
+    votes: 19232,
+    running_times: 5640,
+  },
+  {
+    title: "101 Dalmatians",
+    year: 1996,
+    rating: 5.7,
+    votes: 83245,
+    running_times: 6180,
+  },
+];
+// Create an array of movies containing the movies with a short title
+const shortMovieTitle = movies.filter((movies) => movies.title.length < 10);
+console.log(shortMovieTitle);
+// Create an array of movie titles with long movie titles
+const longMovieTitle = movies.filter((movies) => movies.title.length > 10);
+console.log(longMovieTitle);
+
+// Count the number of movies made between 1980-1989 (including both the years)
+
+const moviesMadeByYear = movies.filter(
+  (movies) => movies.year >= 1980 && movies.year <= 1989
+);
+console.log(moviesMadeByYear);
+
+// Create a new array that has an extra key called tag. The tag is based on the rating: Good (>= 7), Average (>= 4 and < 7), Bad (< 4)
+
+movies.map((movies) => {
+  if (movies.rating >= 7) movies.tag = "good";
+  else if (movies.rating < 7 && movies.rating >= 4) movies.tag = "Average";
+  else movies.tag = "Bad";
+});
+
+console.log(movies);
+
+// Using chaining, first filter the movies array to only contain the movies rated higher than 6. Now map the movies array to only the rating of the movies.
+// const moviesHighRatings = movies.filter((movies) => movies.rating > 6);
+// const mapRatings = movies.map((movies) => movies.rating);
+
+const moviesHighRatings = movies
+  .filter((movies) => movies.rating > 6)
+  .map((movies) => movies.rating);
+console.log(moviesHighRatings);
+
+// Count the total number of movies containing any of following keywords: Surfer, Alien or Benjamin. So if there were 3 movies that contained Surfer, 1 with Alien and 2 with Benjamin, you would return 6. Can you make sure the search is case insensitive?
+
+const totalNumberOfMovies = movies.filter(
+  (movies) =>
+    movies.title.toLowerCase().includes("alien") ||
+    movies.title.toLowerCase().includes("surfer") ||
+    movies.title.toLowerCase().includes("Benjamin")
+);
+console.log(totalNumberOfMovies);
+
+// Create an array of movies where a word in the title is duplicated.
+const moviesWithDublicateName = movies.filter(
+  (movies) => movies.title.indexOf(movies.title) === movies.title
+);
+console.log(moviesWithDublicateName);
+
+// Calculate the average rating of all the movies using reduce. Optional
+const averageRating = movies.map((movies) => movies.rating);
+const reducer = (accumulator, currentValue) => accumulator + currentValue;
+console.log(averageRating.reduce(reducer));
```

Comment:
Average means something like `avg = (a1+a2...+an)/n` where n is the number of elements.
```suggestion
console.log(averageRating.reduce(reducer));
```